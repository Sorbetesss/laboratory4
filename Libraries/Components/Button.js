/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @providesModule Button
 * @flow
 */
'use strict';

const ColorPropType = require('ColorPropType');
const Platform = require('Platform');
const React = require('React');
const StyleSheet = require('StyleSheet');
const Text = require('Text');
const TouchableNativeFeedback = require('TouchableNativeFeedback');
const TouchableOpacity = require('TouchableOpacity');
const View = require('View');

const invariant = require('fbjs/lib/invariant');

/**
 * A basic button component that should render nicely on any platform. Supports
 * a minimal level of customization.
 *
 * <center><img src="img/buttonExample.png"></img></center>
 *
 * > You can change the style of the button using the props `styles`.
 *
 * If this button doesn't look right for your app, you can build your own
 * button using [TouchableOpacity](https://facebook.github.io/react-native/docs/touchableopacity.html)
 * or [TouchableNativeFeedback](https://facebook.github.io/react-native/docs/touchablenativefeedback.html).
 * For inspiration, look at the [source code for this button component](https://github.com/facebook/react-native/blob/master/Libraries/Components/Button.js).
 * Or, take a look at the [wide variety of button components built by the community](https://js.coach/react-native?search=button).
 *
 * Example usage:
 *
 * ```
 * <Button
 *   onPress={onPressLearnMore}
 *   title="Learn More"
 *   color="#841584"
 *   accessibilityLabel="Learn more about this purple button"
 * />
 * ```
 *
 * Example of usage with a custom style.
 *
 * ```
 * <Button
 *   onPress={onPressLearnMore}
 *   title="Learn More"
 *   color="#841584"
 *   accessibilityLabel="Learn more about this purple button"
 *   styles={{
 *     text: Platform.select({
 *       ios: {
 *         textDecorationLine: 'underline',
 *       },
 *       android: {
 *        textDecorationLine: 'underline',
 *       },
 *     })
 *   }}
 * />
 * ```
 *
 */

class Button extends React.Component {

  props: {
    title: string,
    onPress: () => any,
    color?: ?string,
    accessibilityLabel?: ?string,
    disabled?: ?boolean,
    styles: shape,
  };

  static propTypes = {
    /**
     * Text to display inside the button
     */
    title: React.PropTypes.string.isRequired,
    /**
     * Text to display for blindness accessibility features
     */
    accessibilityLabel: React.PropTypes.string,
    /**
     * Color of the text (iOS), or background color of the button (Android)
     */
    color: ColorPropType,
    /**
     * If true, disable all interactions for this component.
     */
    disabled: React.PropTypes.bool,
    /**
     * Handler to be called when the user taps the button
     */
    onPress: React.PropTypes.func.isRequired,
    /**
     * The object with the styles to apply to the button. The styles should follow the
     * [current button styles format](https://github.com/facebook/react-native/blob/master/Libraries/Components/Button.js#L141).
     */
    styles: React.PropTypes.shape({
      button: React.PropTypes.object,
      text: React.PropTypes.object,
      buttonDisabled: React.PropTypes.object,
      textDisabled: React.PropTypes.object,
    }),
  };

  render() {
    const {
      accessibilityLabel,
      color,
      onPress,
      title,
      disabled,
    } = this.props;
    const buttonStyles = [styles.button, {...this.props.styles.button}];
    const textStyles = [styles.text, {...this.props.styles.text}];
    const Touchable = Platform.OS === 'android' ? TouchableNativeFeedback : TouchableOpacity;
    if (color && Platform.OS === 'ios') {
      textStyles.push({color: color});
    } else if (color) {
      buttonStyles.push({backgroundColor: color});
    }
    if (disabled) {
      buttonStyles.push([styles.buttonDisabled, {...this.props.styles.buttonDisabled}]);
      textStyles.push([styles.textDisabled, {...this.props.styles.textDisabled}]);
    }
    invariant(
      typeof title === 'string',
      'The title prop of a Button must be a string',
    );
    const formattedTitle = Platform.OS === 'android' ? title.toUpperCase() : title;
    return (
      <Touchable
        accessibilityComponentType="button"
        accessibilityLabel={accessibilityLabel}
        accessibilityTraits={['button']}
        disabled={disabled}
        onPress={onPress}>
        <View style={buttonStyles}>
          <Text style={textStyles}>{formattedTitle}</Text>
        </View>
      </Touchable>
    );
  }
}

// Material design blue from https://material.google.com/style/color.html#color-color-palette
let defaultBlue = '#2196F3';
if (Platform.OS === 'ios') {
  // Measured default tintColor from iOS 10
  defaultBlue = '#0C42FD';
}

const styles = StyleSheet.create({
  button: Platform.select({
    ios: {},
    android: {
      elevation: 4,
      backgroundColor: defaultBlue,
      borderRadius: 2,
    },
  }),
  text: Platform.select({
    ios: {
      color: defaultBlue,
      textAlign: 'center',
      padding: 8,
      fontSize: 18,
    },
    android: {
      textAlign: 'center',
      color: 'white',
      padding: 8,
      fontWeight: '500',
    },
  }),
  buttonDisabled: Platform.select({
    ios: {},
    android: {
      elevation: 0,
      backgroundColor: '#dfdfdf',
    }
  }),
  textDisabled: Platform.select({
    ios: {
      color: '#cdcdcd',
    },
    android: {
      color: '#a1a1a1',
    }
  }),
});

module.exports = Button;
