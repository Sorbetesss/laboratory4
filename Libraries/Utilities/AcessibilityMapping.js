/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

// Map role values to AccessibilityRole values
export const roleToAccessibilityRoleMapping = {
  alert: 'alert',
  alertdialog: undefined,
  application: undefined,
  article: undefined,
  banner: undefined,
  button: 'button',
  cell: undefined,
  checkbox: 'checkbox',
  columnheader: undefined,
  combobox: 'combobox',
  complementary: undefined,
  contentinfo: undefined,
  definition: undefined,
  dialog: undefined,
  directory: undefined,
  document: undefined,
  feed: undefined,
  figure: undefined,
  form: undefined,
  grid: 'grid',
  group: undefined,
  heading: 'header',
  img: 'image',
  link: 'link',
  list: 'list',
  listitem: undefined,
  log: undefined,
  main: undefined,
  marquee: undefined,
  math: undefined,
  menu: 'menu',
  menubar: 'menubar',
  menuitem: 'menuitem',
  meter: undefined,
  navigation: undefined,
  none: 'none',
  note: undefined,
  presentation: 'none',
  progressbar: 'progressbar',
  radio: 'radio',
  radiogroup: 'radiogroup',
  region: undefined,
  row: undefined,
  rowgroup: undefined,
  rowheader: undefined,
  scrollbar: 'scrollbar',
  searchbox: 'search',
  separator: undefined,
  slider: 'adjustable',
  spinbutton: 'spinbutton',
  status: undefined,
  summary: 'summary',
  switch: 'switch',
  tab: 'tab',
  table: undefined,
  tablist: 'tablist',
  tabpanel: undefined,
  term: undefined,
  timer: 'timer',
  toolbar: 'toolbar',
  tooltip: undefined,
  tree: undefined,
  treegrid: undefined,
  treeitem: undefined,
};
