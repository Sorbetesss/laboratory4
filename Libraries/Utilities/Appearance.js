/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow
 */

'use strict';

import EventEmitter from '../vendor/emitter/EventEmitter';
import NativeEventEmitter from '../EventEmitter/NativeEventEmitter';
import NativeAppearance, {type AppearancePreferences} from './NativeAppearance';
import invariant from 'invariant';

const COLOR_SCHEME_NAME = {
  light: 'light',
  dark: 'dark',
  noPreference: 'no-preference',
};
type ColorSchemeName = $Keys<{
  light: string,
  dark: string,
  noPreference: string,
}>;
type AppearanceListener = (preferences: AppearancePreferences) => void;

const eventEmitter = new EventEmitter();

let appearancePreferencesInitialized = false;
let appearancePreferences: AppearancePreferences;

class Appearance {
  /**
   * Note: Although appearance is available immediately, it may change (e.g
   * Dark Mode) so any rendering logic or styles that depend on this should try
   * to call this function on every render, rather than caching the value (for
   * example, using inline styles rather than setting a value in a
   * `StyleSheet`).
   *
   * Example: `const colorScheme = Appearance.get('colorScheme');`
   *
   * @param {string} preference Name of preference (e.g. 'colorScheme').
   * @returns {ColorSchemeName} Value for the preference.
   */
  static get(preference: string): ColorSchemeName {
    invariant(
      appearancePreferences[preference],
      'No preference set for key ' + preference,
    );
    return appearancePreferences[preference];
  }

  /**
   * This should only be called from native code by sending the
   * appearanceChanged event.
   *
   * @param {object} appearancePreferences Simple string-keyed object of
   * appearance preferences to set.
   */
  static set(preferences: AppearancePreferences): void {
    let {colorScheme} = preferences;
    appearancePreferences = {colorScheme};

    if (appearancePreferencesInitialized) {
      // Don't fire 'change' the first time the dimensions are set.
      eventEmitter.emit('change', preferences);
    } else {
      appearancePreferencesInitialized = true;
    }
  }

  /**
   * Add an event handler that is fired when appearance preferences change.
   */
  static addChangeListener(listener: AppearanceListener): void {
    eventEmitter.addListener('change', listener);
  }

  /**
   * Remove an event handler.
   */
  static removeChangeListener(listener: AppearanceListener): void {
    eventEmitter.removeListener('change', listener);
  }
}

if (NativeAppearance) {
  const nativeEventEmitter = new NativeEventEmitter(NativeAppearance);
  // Subscribe before calling to make sure we don't miss any updates in between.
  nativeEventEmitter.addListener(
    'appearanceChanged',
    (newAppearance: AppearancePreferences) => {
      Appearance.set(newAppearance);
    },
  );
  Appearance.set(NativeAppearance.getPreferences());
} else {
  Appearance.set({colorScheme: 'no-preference'});
}

module.exports = Appearance;
