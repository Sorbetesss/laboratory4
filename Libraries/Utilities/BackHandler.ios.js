/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * On Apple TV, this implements back navigation using the TV remote's menu button.
 * On iOS, this just implements a stub.
 *
 * @providesModule BackHandler
 */

'use strict';

var Platform = require('Platform');
var TVEventHandler = require('TVEventHandler');

type BackPressEventName = $Enum<{
  backPress: string,
}>;

function emptyFunction() {}

var BackHandler;

if (Platform.isTVOS) {

  var _tvEventHandler = new TVEventHandler();

  var _backPressSubscriptions = new Set();

  _tvEventHandler.enable(this, function(cmp, evt) {
    if (evt && evt.eventType === 'menu') {
      var backPressSubscriptions = new Set(_backPressSubscriptions);
      var invokeDefault = true;
      var subscriptions = [...backPressSubscriptions].reverse();
      for (var i = 0; i < subscriptions.length; ++i) {
        if (subscriptions[i]()) {
          invokeDefault = false;
          break;
        }
      }

      if (invokeDefault) {
        BackHandler.exitApp();
      }
    }
  });

  BackHandler = {
    exitApp: emptyFunction,

    addEventListener: function (
      eventName: BackPressEventName,
      handler: Function
    ): {remove: () => void} {
      _backPressSubscriptions.add(handler);
      return {
        remove: () => BackHandler.removeEventListener(eventName, handler),
      };
    },

    removeEventListener: function(
      eventName: BackPressEventName,
      handler: Function
    ): void {
      _backPressSubscriptions.delete(handler);
    },

  };


} else {


  BackHandler = {
    exitApp: emptyFunction,
    addEventListener() {
      return {
        remove: emptyFunction,
      };
    },
    removeEventListener: emptyFunction,
  };


}

module.exports = BackHandler;

