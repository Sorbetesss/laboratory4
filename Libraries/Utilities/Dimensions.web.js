/**
 * @providesModule Dimensions
 */
'use strict';

var invariant = require('invariant');

var dimensions = {

    window: {
        scale: window.devicePixelRatio,
        fontScale: window.devicePixelRatio,
    },

};

var updateWindowSize = function() {
    dimensions.window.width = window.innerWidth;
    dimensions.window.height = window.innerHeight;
};
updateWindowSize();
window.addEventListener('resize', updateWindowSize);

class Dimensions {
  
    static set(dims: {[key:string]: any}): bool {
        Object.assign(dimensions, dims);
        return true;
    }

    static get(dim: string): Object {
        invariant(dimensions[dim], 'No dimension set for key ' + dim);
        return dimensions[dim];
    }

}

module.exports = Dimensions;
