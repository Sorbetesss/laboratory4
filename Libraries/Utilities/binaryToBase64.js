/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule binaryToBase64
 * @flow
 */
'use strict';

/* $FlowFixMe(>=0.54.0 site=react_native_oss) This comment suppresses an error
 * found when Flow v0.54 was deployed. To see the error delete this comment and
 * run Flow. */
const base64 = require('base64-js');

function binaryToBase64(data: ArrayBuffer | $ArrayBufferView) {
  let dataBuffer = data;
  if (dataBuffer instanceof ArrayBuffer) {
    dataBuffer = new Uint8Array(dataBuffer);
  }
  if (dataBuffer instanceof Uint8Array) {
    return base64.fromByteArray(dataBuffer);
  }
  if (!ArrayBuffer.isView(dataBuffer)) {
    throw new Error('data must be ArrayBuffer or typed array');
  }
  const {buffer, byteOffset, byteLength} = dataBuffer;
  return base64.fromByteArray(new Uint8Array(buffer, byteOffset, byteLength));
}

module.exports = binaryToBase64;
