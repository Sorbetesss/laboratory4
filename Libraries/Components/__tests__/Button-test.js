import * as React from 'react';

import Button from '../Button';
import {expectRendersMatchingSnapshot} from '../../Utilities/ReactNativeTestTools';

describe('<Button />', () => {
  it('should render as expected', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});

describe('<Button />', () => {
  it('should be disabled when disabled={true}', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" disabled={true} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});

describe('<Button />', () => {
  it('should be disabled when disabled={true} and accessibilityState={{disabled: true}}', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" disabled={true} accessibilityState={{disabled: true}} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});

describe('<Button />', () => {
  it('should be disabled when disabled is empty and accessibilityState={{disabled: true}}', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" accessibilityState={{disabled: true}} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});



describe('<Button />', () => {
  it('should overwrite accessibilityState with value of disabled prop', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" disabled={true} accessibilityState={{disabled: false}} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});

describe('<Button />', () => {
  it('should not be disabled when disabled={false} and accessibilityState={{disabled: true}}', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" disabled={false} accessibilityState={{disabled: true}} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});

describe('<Button />', () => {
  it('should not be disabled when disabled={false} and accessibilityState={{disabled: false}}', () => {
    expectRendersMatchingSnapshot(
      'Button',
      () => (
        <Button title="Test Button" disabled={false} accessibilityState={{disabled: false}} />
      ),
      () => {
        jest.dontMock('../Button');
      },
    );
  });
});
