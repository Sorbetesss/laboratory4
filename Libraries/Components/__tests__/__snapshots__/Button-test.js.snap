// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be disabled when disabled is empty and accessibilityState={{disabled: true}}: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled is empty and accessibilityState={{disabled: true}}: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled is empty and accessibilityState={{disabled: true}}: should shallow render as <Button /> when mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  title="Test Button"
/>
`;

exports[`<Button /> should be disabled when disabled is empty and accessibilityState={{disabled: true}}: should shallow render as <Button /> when not mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  title="Test Button"
/>
`;

exports[`<Button /> should be disabled when disabled={true} and accessibilityState={{disabled: true}}: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled={true} and accessibilityState={{disabled: true}}: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled={true} and accessibilityState={{disabled: true}}: should shallow render as <Button /> when mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should be disabled when disabled={true} and accessibilityState={{disabled: true}}: should shallow render as <Button /> when not mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should be disabled when disabled={true}: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled={true}: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should be disabled when disabled={true}: should shallow render as <Button /> when mocked 1`] = `
<Button
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should be disabled when disabled={true}: should shallow render as <Button /> when not mocked 1`] = `
<Button
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: false}}: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: false}}: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: false}}: should shallow render as <Button /> when mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={false}
  title="Test Button"
/>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: false}}: should shallow render as <Button /> when not mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={false}
  title="Test Button"
/>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: true}}: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: true}}: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      disabled={false}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: true}}: should shallow render as <Button /> when mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={false}
  title="Test Button"
/>
`;

exports[`<Button /> should not be disabled when disabled={false} and accessibilityState={{disabled: true}}: should shallow render as <Button /> when not mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={false}
  title="Test Button"
/>
`;

exports[`<Button /> should overwrite accessibilityState with value of disabled prop: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should overwrite accessibilityState with value of disabled prop: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      disabled={true}
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#cdcdcd",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should overwrite accessibilityState with value of disabled prop: should shallow render as <Button /> when mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should overwrite accessibilityState with value of disabled prop: should shallow render as <Button /> when not mocked 1`] = `
<Button
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={true}
  title="Test Button"
/>
`;

exports[`<Button /> should render as expected: should deep render when mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should render as expected: should deep render when not mocked (please verify output manually) 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;

exports[`<Button /> should render as expected: should shallow render as <Button /> when mocked 1`] = `
<Button
  title="Test Button"
/>
`;

exports[`<Button /> should render as expected: should shallow render as <Button /> when not mocked 1`] = `
<Button
  title="Test Button"
/>
`;
